packaging/rpm/paack.py

```diff
@@ -0,0 +1,345 @@
+                # directories, hard links and regular files
+                output += "%%attr(%o,%d,%d) \"%%{imageStore}%s\"\n" % (info.mode, info.uid, info.gid, info.name)
+            if info.type in [tarfile.SYMTYPE]:
+                # symlinks have no mode permissions
+                output += "%%attr(-,%d,%d) \"%%{imageStore}%s\"\n" % (info.uid, info.gid, info.name)
+            if info.type in [tarfile.DIRTYPE]:
+                # directories, hard links and regular files
+                output += "%%dir %%attr(%o,%d,%d) \"%%{imageStore}%s\"\n" % (info.mode, info.uid, info.gid, info.name)
+
+        self._files_data += "\n\n%ifarch " + arch + "\n" + output + "%endif\n"
+        self._extract_archs += "\n\n%ifarch " + arch + "\n" + "tar xfj %{SOURCE"+str(self._source_i) + "}\n%endif"
+        self._sources += "Source%d: %s\n" % (self._source_i, os.path.basename(tar))
+        self._source_i += 1
+
+
+class PaackYaml(object):
+    def __init__(self, filename):
+        with open(filename, "r") as f:
+            try:
+                self._data = yaml.safe_load(f)
+            except yaml.YAMLError as exc:
+                print(exc)
+                sys.exit(1)
+        self.check()
+
+    def data(self):
+        return self._data # TODO: instead of this, provide methods to traverse the content, this interface is too loose
+
+    def check(self):
+        pass # TODO
+
+
+def system(cmd):
+    print(YELLOW + "> " + cmd + CLEAR)
+    result = os.system(cmd)
+    if result != 0:
+        print(RED + "command failed" + CLEAR)
+        sys.exit(result)
+    return result
+
+
+
+class SRPMBuilderCommand(object):
+    def __init__(self, args):
+        self._rpmbuild_dir = args.rpmbuild_dir.replace("~", os.environ.get("HOME"))
+        self._tmp = args.tmp
+        self._yaml = args.yaml
+        self._no_cleanup = args.no_cleanup
+
+    def build(self):
+        return self._build_from_yaml()
+
+    def _build_from_yaml(self):
+        data = PaackYaml(self._yaml).data()
+
+        # ensure output directory
+        system("mkdir -p " + os.path.join(self._rpmbuild_dir, 'SPECS'))
+
+        result_files = []
+
+        for package in data['packages']:
+            out_srpm = os.path.join(self._rpmbuild_dir, 'SRPMS',
+                                    '%s-%s-%s.src.rpm' % (package['name'], package['version'], package['release']))
+
+            # if we are in no-cleanup mode the existing srpm is reused instead
+            if not (os.path.exists(out_srpm) and self._no_cleanup):
+                spec = SpecFile(package['name'], os.path.join(self._rpmbuild_dir, 'SPECS',
+                    '%s-%s-%s.spec' % (package['name'], package['version'], package['release'])))
+                for arch_info in package['arch']:
+                    arch = arch_info['name']
+                    tar_path = self._create_tarball(package, arch_info)
+                    print("scanning %s to generate files section for arch %s" %(tar_path, arch))
```

Comment:
```suggestion
                    print("scanning %s to generate files section for arch %s" % (tar_path, arch))
```
nit: space for style consistency